ejb/src/main/java/org/wildfly/httpclient/ejb/HttpEJBDiscoveryProvider.java

```diff
@@ -159,12 +160,14 @@ private void discoverFromConnection(final EJBClientConnection connection, final
                 ((result, response, closeable) -> {
                     try {
                         final Unmarshaller unmarshaller = targetContext.getHttpMarshallerFactory(request).createUnmarshaller();
-
-                        unmarshaller.start(new InputStreamByteInput(result));
-                        int size = unmarshaller.readInt();
-
-                        for (int i = 0; i < size; i++) {
-                            EJBModuleIdentifier ejbModuleIdentifier = (EJBModuleIdentifier) unmarshaller.readObject();
+                        final ByteInput in = new InputStreamByteInput(result);
```

Comment:
try with resources?

Reply:
This code moved to ClientHandlers. It is fixed there in the way you suggested.